channels/audin/client/alsa/audin_alsa.c

```diff
@@ -169,17 +174,20 @@ static DWORD WINAPI audin_alsa_thread_func(LPVOID arg)
 
 		if (err == -EPIPE)
 		{
-			snd_pcm_recover(capture_handle, (int)err, 0);
+			const int rc = snd_pcm_recover(capture_handle, (int)err, 0);
+			if (rc < 0)
+				WLog_Print(alsa->log, WLOG_WARN, "snd_pcm_recover (%s)", snd_strerror(rc));
+
 			continue;
 		}
 		else if (err < 0)
 		{
-			WLog_Print(alsa->log, WLOG_ERROR, "snd_pcm_readi (%s)", snd_strerror(error));
+			WLog_Print(alsa->log, WLOG_ERROR, "snd_pcm_readi (%s)", snd_strerror(err));
 			error = ERROR_INTERNAL_ERROR;
 			break;
 		}
 
-		error = alsa->receive(&alsa->aformat, buffer, (long)error * alsa->bytes_per_frame,
+		error = alsa->receive(&alsa->aformat, buffer, (long)err * alsa->bytes_per_frame,
```

Comment:
Especially in this context the name `err` seems to be inappropriate (and also kind of caused  this issue) - maybe it's better to rename it something what it really is: "a positive number of frames actually read"